### PR TITLE
[Tooltip]: Add floating UI, a tooltip & convert the Dropdown to use the new floating component

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
@petemill & @AlanBreck I think `floating-ui` is going to be super helpful going forward :smile:

### CODE REVIEW

File: examples/example-ui-react/src/App.tsx
Comment (petemill) [inline examples/example-ui-react/src/App.tsx:14]: Is this for testing for errors? Although I'm a bit confused what `f` is used for 😄

File: package.json
Comment (petemill) [inline package.json:9]: let's get a security review for this

**Review by petemill — COMMENTED** [review]
one thing I'm a little confused about is that the dropdown in storybook doesn't seem to be flipping to opening above the "input" if there isn't any room underneath in the viewport (or the document). Am I understanding flip and its default value correctly?

image

File: src/components/tooltip/tooltip.svelte
Comment (petemill) [inline src/components/tooltip/tooltip.svelte:90]: Can we add aria attributes for hidden please

Comment (petemill) [reply src/components/tooltip/tooltip.svelte:90]: Actually maybe we don't need that because the `hidden` attribute seems to be making the content hidden from the accessibility tree

File: src/components/dropdown/dropdown.svelte
Comment (petemill) [inline src/components/dropdown/dropdown.svelte:171]: This isn't getting updated when dropdown width changes. Make storybook window narrower, then click it open and popup will be too wide.

Comment (nullhook) [reply src/components/dropdown/dropdown.svelte:171]: image

Also, the content overflows. Perhaps, add shadow, rounding and border in the outerbox?

File: package.json
Comment (fallaciousreasoning) [reply package.json:9]: https://github.com/brave/reviews/issues/1400

File: examples/example-ui-react/src/App.tsx
Comment (fallaciousreasoning) [reply examples/example-ui-react/src/App.tsx:14]: Whoops, I was probably using it to see what colors are available :flushed:

File: src/components/dropdown/dropdown.svelte
Comment (fallaciousreasoning) [reply src/components/dropdown/dropdown.svelte]: @nullhook I'm tempted to leave that as a follow up, because I'm going on PTO next week, and it's like that even without this PR - we should definitely fix it though

**Review by petemill — APPROVED** [review]

**Comment by fallaciousreasoning** [comment]
Blocked on security review https://github.com/brave/reviews/issues/1400

File: src/components/alert/alertCenter.svelte
Comment (petemill) [inline src/components/alert/alertCenter.svelte:94]: cc @fallaciousreasoning this seems needed to use in projects that don't know about `??` (like brave-core's webpack v4!) as I don't think typescript sees these svelte parts, only the svelte compiler

**Comment by petemill** [comment]
We can make size middleware work as a follow-up https://github.com/brave/leo/issues/391

File: src/components/dropdown/dropdown.stories.svelte
Comment (petemill) [inline src/components/dropdown/dropdown.stories.svelte:56]: nit: a "context menu" is the whole menu. I would suggest that dropdown menu should only support leo-option, and in ButtonMenu story we can suggest both

File: src/components/buttonMenu/buttonMenu.stories.svelte
Comment (petemill) [inline src/components/buttonMenu/buttonMenu.stories.svelte:37]:
```suggestion
    description="The ButtonMenu provides several slots for customization. It accepts '<leo-menu-item>' and '<leo-option>' elements as well as any other element types. Use '<leo-menu-item>' for menu-style commands and '<leo-option>' for select-style selectable choices."
```

Comment (nullhook) [reply src/components/buttonMenu/buttonMenu.stories.svelte]: Fixed

File: src/components/alert/alertCenter.svelte
Comment (fallaciousreasoning) [reply src/components/alert/alertCenter.svelte:94]: :facepalm:

**Pull request merged**
